### PR TITLE
add consumer test for e2e

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>

Add the e2e test case for CURD on consumers

### CODE REVIEW

File: test/e2e/pkg/suite_test.go [hunk 1]
```diff
@@ -19,40 +20,32 @@ import (
 )
 
 var (
+	consumerOpts     *ConsumerOptions
 	apiServerAddress string
-	kubeconfig       string
-	consumer_name    string
-	kubeClient       *kubernetes.Clientset
 	apiClient        *openapi.APIClient
 	helper           *test.Helper
```
Comment (morvencao) [inline test/e2e/pkg/suite_test.go:26]: @clyang82 
I'm curious why we need `helper` in e2e testing? It's mainly used by integration test to set up maestro servers, but for e2e testing, the only use of `helper` is to create openapi resource?

File: Makefile [hunk 2]
```diff
@@ -406,8 +406,14 @@ e2e-test/teardown:
 	./test/e2e/setup/e2e_teardown.sh
 .PHONY: e2e-test/teardown
 
-e2e-test: e2e-test/teardown e2e-test/setup
+e2e-test:
+	@if [ -z "$$consumer_name" ] || [ -z "$$api_server" ] || [ -z "$$consumer_kubeconfig" ]; then \
```
Comment (clyang82) [inline Makefile:410]: why need this judgement?

File: test/e2e/pkg/suite_test.go (hunk 1 above)
Comment (clyang82) [reply test/e2e/pkg/suite_test.go:26]: Yes. only use it to create openapi resource.

File: test/e2e/pkg/consumer_test.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+			Expect(consumerList).NotTo(BeNil())
+			Expect(len(consumerList.Items) > 0).To(BeTrue())
+
+			got := false
+			for _, consumer := range consumerList.Items {
+				if *consumer.Name == consumerOpts.Name {
+					got = true
+				}
+			}
+			Expect(got).To(BeTrue())
```
Comment (clyang82) [inline test/e2e/pkg/consumer_test.go:34]: how to guarantee the consumer is there? the test order is random. if run this test at the beginning, the consumer should be example. right?

File: Makefile (hunk 2 above)
Comment (yanmxa) [reply Makefile:410]: By default, the e2e test will run on the KinD cluster by running `e2e-test/setup`. Here we can provide a way to run the  `e2e-test` on other platforms like `AKS` or `Openshift`. All we need to do is given the environment variables `consumer_name`, `api_server`, and `consumer_kubeconfig`.

I found it convenient during testing, but could this be considered over-engineering? Do you have any suggestions?

File: test/e2e/pkg/consumer_test.go (hunk 3 above)
Comment (yanmxa) [reply test/e2e/pkg/consumer_test.go:34]: For this case, we already have generated a consumer before running the e2e test. So the consumer must be there~

File: Makefile (hunk 2 above)
Comment (clyang82) [reply Makefile:413]: maybe export a single environment to indicate whether need to setup the environment or use the existing environment. What do you think?

File: test/e2e/pkg/consumer_test.go (hunk 3 above)
Comment (clyang82) [reply test/e2e/pkg/consumer_test.go:34]: How about the case you want to run it in an existing environment? Do you require create the consumer as a pre-requirement? maybe just adjust your order to create/get/patch

**Review by clyang82 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (yanmxa) [reply Makefile]: Maybe we can add this feature in another PR, I removed it from the current.

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**